confluent_kafka/avro/cached_schema_registry_client.py

```diff
@@ -366,67 +372,62 @@ def test_compatibility(self, subject, avro_schema, version='latest'):
         try:
             result, code = self._send_request(url, method='POST', body=body)
             if code == 404:
-                log.error(("Subject or version not found:" + str(code)))
+                log.error("Subject or version not found. Error code: {}".format(code))
                 return False
             elif code == 422:
-                log.error(("Invalid subject or schema:" + str(code)))
+                log.error("Invalid subject or schema. Error code: {}".format(code))
                 return False
-            elif code >= 200 and code <= 299:
+            elif 200 <= code <= 299:
                 return result.get('is_compatible')
             else:
-                log.error("Unable to check the compatibility: " + str(code))
+                log.error("Unable to check the compatibility. Error code: {}".format(code))
                 return False
         except Exception as e:
             log.error("_send_request() failed: %s", e)
             return False
 
-    def update_compatibility(self, level, subject=None):
+    def update_compatibility(self, level):
         """
-        PUT /config/(string: subject)
+        PUT /config
 
-        Update the compatibility level for a subject.  Level must be one of:
+        Update the compatibility level for a subject.
 
         :param str level: ex: 'NONE','FULL','FORWARD', or 'BACKWARD'
+        :raises: ClientError: if request was unsuccessful or an invalid compatibility level was provided
+        :returns: new compatibility level, one of 'NONE','FULL','FORWARD', or 'BACKWARD'
+        :rtype: str
         """
         if level not in VALID_LEVELS:
-            raise ClientError("Invalid level specified: %s" % (str(level)))
+            raise ClientError("Invalid level specified: {}".format(level))
 
         url = '/'.join([self.url, 'config'])
-        if subject:
-            url += '/' + subject
 
         body = {"compatibility": level}
         result, code = self._send_request(url, method='PUT', body=body)
-        if code >= 200 and code <= 299:
+        if 200 <= code <= 299:
             return result['compatibility']
         else:
-            raise ClientError("Unable to update level: %s. Error code: %d" % (str(level)), code)
+            raise ClientError("Unable to update level: {}".format(level), code)
 
-    def get_compatibility(self, subject=None):
```

Comment:
This too is a valid Schema Registry operation at the subject level.

https://docs.confluent.io/current/schema-registry/docs/api.html#get--config-(string-%20subject)